### PR TITLE
Deprecate TimedItem events and implement AnimationPlayer events

### PR DESCRIPTION
**Pull request by alancutter**
This change deprecates the event model for TimedItem (onstart, oniteration, onend, oncancel).
AnimationPlayers now have an onfinish event as an alternative.
Specification: http://dev.w3.org/fxtf/web-animations/#dfn-player-event

### CODE REVIEW

File: test/testcases/test-player-addeventlistener.html [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+var testH = async_test('Check onfinish handler ordering after removal');
+player.addEventListener('finish', function() {
+  testH.step(function() { assert_equals(onEventOrderingAfterRemoval, 'abc'); });
+  testH.done();
+});
+
+player = createPlayer();
+var onEventDuplication = 0;
+var duplicateFunction = function() { onEventDuplication += 1; };
+player.addEventListener('finish', duplicateFunction);
```
Comment (ewilligers) [inline test/testcases/test-player-addeventlistener.html]: Perhaps also have a test where duplicateFunction is passed to addEventListener twice

Comment (alancutter) [inline test/testcases/test-player-addeventlistener.html]: Done.

File: test/testcases/test-player-finish-event.html [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    }
+    i++;
+  });
+}
+
+var player;
+
+player = createPlayer();
+assertFinishEvents('Players should fire finished events when they complete', player, [
+  { timelineTime: 2, currentTime: 2 },
```
Comment (dstoc) [inline test/testcases/test-player-finish-event.html]: In Google style we would normally omit the spaces in a single line object initializer:

```
{timelineTime: 2, currentTime: 2},
```

File: test/testcases/test-player-finish-event.html [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+]);
+
+player = createPlayer();
+player.source = null;
+assertFinishEvents('A player with no source should report finished immediately', player, [
```
Comment (dstoc) [inline test/testcases/test-player-finish-event.html]: immediately -> on the next sample?

**Comment by dstoc** [comment]
lgtm

File: test/testcases/test-player-finish-event.html (hunk 2 above)
Comment (alancutter) [inline test/testcases/test-player-finish-event.html]: Done.

File: test/testcases/test-player-finish-event.html (hunk 3 above)
Comment (alancutter) [inline test/testcases/test-player-finish-event.html]: Done.

**Pull request merged**